### DISCUSSION TITLE
error calling TypeDeclaration: Struct:"blkgq_entry": field 0: type *btf.Pointer: not supported

### DISCUSSION THREAD

**Discussion by alahaiyo** [cilium/ebpf#750] [Q&A] [answered]
blkgq_entry is a structure defined in my c code ：
struct blkgq_entry {
	struct blkcg_gq *blkgq;
	struct cgroup *cgroup;
	struct gendisk *disk;
}
go:generate：
//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -cc clang FreezeCount ../../../../bpf/io/freeze_count.c -- -I../../../.    ./bpf/include -D__TARGET_ARCH_x86

Run _go generate -x_ after which an error will be reported:
Error: can't write /root/freezecount_bpfeb.go: can't generate types: template: common:19:4: executing "common" at <$.TypeDeclaration>: error calling TypeDeclaration: Struct:"blkgq_entry": field 0: type *btf.Pointer: not supported

I remember that on ebpfv0.7.0 it was fine, on ebpfv0.9.0 it reports an error

**Comment by lmb** [comment, marked as answer]
Can you double check that old versions worked? I don't think we ever supported generating pointer types.

What do you want to do with `blkgq_entry->disk` in Go? There is no way to share pointers between Go and eBPF, which is why you currently can't generate a type for such structs.

P.S.: Please pass `-target amd64` to `bpf2go` instead of `-D__TARGET_ARCH_x86`.

**Reply by alahaiyo** [reply]
Thanks! I use the value of the pointer as the key and I should have defined this as u64.
